src/commonTest/kotlin/it/krzeminski/snakeyaml/engine/kmp/CommonTestResourcesAccess.kt

```diff
@@ -5,19 +5,28 @@ import okio.ByteString
 
 /**
  * Retrieves a string content from common resources using a given path.
- * TODO: Add proper error handling if the resource doesn't exist.
  *
  * @param path The slash-delimited path to the resource.
  * @return The content of the resource as a UTF-8 encoded string with normalized line breaks.
+ * @throws IllegalArgumentException if the resource doesn't exist.
  */
 fun stringFromResources(path: String): String {
     val segments = path.split("/")
     return (segments
         .dropLast(1)
-        .fold(CommonTestResources.resourcesMap) { map, segment ->
+        .fold(Pair(CommonTestResources.resourcesMap, "/resources/")) { (map, pathSoFar), directory ->
+            require(directory in map) {
```

Comment:
suggestion: we can extract value from the map and check it's existence at the same time. E.g.:

`requireNotNull(map[key]) {} as Map<>'

Reply:
Hmm, I think it would be a bit _too_ smart and unreadable? `requireNotNull(...)` would look weird as the last statement in the lambda, as if the lambda only checked something. That's my impression. I hope you don't mind if we stay with the current version.

Reply:
Of course not) this was just a suggestion